### PR TITLE
refactor: endpoints called by Rasa moved to GraphQL + support for messages metadata

### PR DESCRIPTION
**Pull request by pheel**
- [ ] I wrote tests for the feature
- [ ] I documented the feature if needed

If the feature is a refactor, please explain why your way is better

### CODE REVIEW

File: botfront/imports/api/graphql/config/configResolver.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import yaml from 'js-yaml';
+import { Credentials, Endpoints } from './config.models.js';
+
+export default {
```
Comment (znat) [inline botfront/imports/api/graphql/config/configResolver.js]: It seems to me that it should be a more generic `getProject(s)` query where credentials/endpoints are fields

```
getProject(projectId: "bf') {
   credentials
   endpoints
}
```

File: botfront/imports/api/graphql/botResponses/resolvers/nlgResolver.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import { safeLoad } from 'js-yaml';
+import { sample } from 'lodash';
+import BotResponses from '../botResponses.model';
+
+const subText = (text, slots) => {
```
Comment (znat) [inline botfront/imports/api/graphql/botResponses/resolvers/nlgResolver.js]: May be a more explicit function name that describes what the function does, something like `interpolateSlots`

File: botfront/imports/api/graphql/botResponses/resolvers/nlgResolver.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    // fills in {slotname} in templates
+    const slotSubs = Object.entries(slots).map(s => [`{${s[0]}}`, s[1] || '']);
+    let subbedText = text;
```
Comment (znat) [inline botfront/imports/api/graphql/botResponses/resolvers/nlgResolver.js]: explicit and descriptive full names please: substitutedText

File: botfront/imports/lib/story.utils.js [hunk 4]
```diff
@@ -246,28 +246,15 @@ export const extractDomain = (stories, slots, templates = {}, defaultDomain = {}
...
-        [k.key]: k.values.reduce((vs, v) => ({
-            ...vs,
-            [v.lang]: v.sequence.map(seq => yaml.safeLoad(seq.content)),
-        }), {}),
-    }), {});
-    if (language && language.length > 0) {
-        templates = filterTemplatesByLanguage(templates, language);
-    }
-    return templates;
+    const templates = await newGetBotResponses({ projectId, language });
```
Comment (znat) [inline botfront/imports/lib/story.utils.js]: Why is this called **new**getBotResponses? Is there an old?

File: botfront/imports/lib/story.utils.tests.js [hunk 5]
```diff
@@ -376,3 +552,36 @@ describe('proper flattening of stories', function() {
...
+            expect(response).to.be.deep.equal(
+                responsesExported['en-zz'],
+            );
+        });
+        it('fetch no responses', async () => {
+            const response = await getAllTemplates('test', 'nonexisting');
+            expect(response).to.be.deep.equal({});
+        });
+    });
+}
```
Comment (znat) [inline botfront/imports/lib/story.utils.tests.js]: I think important test cases are missing:
- Testing that the NLG returns the expected response in the expected language
- Testing that the domain is generated as expected

File: botfront/imports/api/graphql/botResponses/resolvers/nlgResolver.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
+import { safeLoad } from 'js-yaml';
+import { sample } from 'lodash';
+import { newGetBotResponses } from '../mongo/botResponses';
+
+const subText = (text, slots) => {
+    // fills in {slotname} in templates
+    const slotSubs = Object.entries(slots).map(s => [`{${s[0]}}`, s[1] || '']);
```
Comment (znat) [inline botfront/imports/api/graphql/botResponses/resolvers/nlgResolver.js]: Create a function and substitute both calls (here and in templates.controller.js)

File: botfront/imports/api/graphql/config/configResolver.js (hunk 1 above)
Comment (pheel) [reply botfront/imports/api/graphql/config/configResolver.js]: can this be in a further iteration?

File: botfront/imports/lib/story.utils.tests.js (hunk 5 above)
Comment (pheel) [reply botfront/imports/lib/story.utils.tests.js]: /* to do:
extractDomain
getStoriesAndDomain
accumulateExceptions
*/

These were never actually covered, all I did is add coverage for getAllTemplates.

Re: Testing that the NLG returns the expected response in the expected language: Yes, we should have tests for our graphql routes.

File: botfront/imports/api/graphql/botResponses/resolvers/nlgResolver.js (hunk 6 above)
Comment (pheel) [reply botfront/imports/api/graphql/botResponses/resolvers/nlgResolver.js]: Can't, since templates.controller.js is in the API package directory.

**Pull request merged**